pkgs/development/tools/pnpm/fetch-deps/pnpm-config-hook.sh

```diff
@@ -23,10 +23,13 @@ pnpmConfigHook() {
     pnpm config set store-dir "$STORE_PATH"
 
     echo "Installing dependencies"
-
-    if [[ -n "$pnpmWorkspace" ]]; then
-        pnpmInstallFlags+=("--filter=$pnpmWorkspace")
+    if [[ -n "$pnpmWorkspaces" ]]; then
+        local IFS=" "
+        for ws in $pnpmWorkspaces; do
```

Comment:
I am a little unsure about this. Shellcheck would definitely warn here, but I assume wrapping this with quotes would break the loop. If only we had structured attrs by default :sweat_smile:

Reply:
just ran shellcheck over this and it doesn't warn. I'm pretty sure I could get away without setting IFS, but it's an additional layer of protection against anything weird happening in the script. IFS does include space by default though, so theoretically it doesn't have to be set, I set it anyways for safety since we don't want to break up args by newlines.

Reply:
I encourage looking at some of the PRs in https://github.com/NixOS/nixpkgs/issues/205690 to get this hook ready for a `__structuredAttrs` world where this will be an array. However, that should not block this.

Reply:
I'll be honest, most of the linked PRs are going right over my head since I don't understand exactly what would need to be changed for `__structuredAttrs` to work properly(or at all)

Reply:
If you comment on that issue I’m sure people would be able to help out making the hook work with and without `__structuerdAttrs`, but I don’t think that needs doing for 24.11; just something that will need doing at some point. (I kind of hope we can default `__structuredAttrs` and `strictDeps` on for 25.05, but it’d be a big push.)

Reply:
I agree with @emilazy here. Let's focus on getting this merged first. `__structuredAttrs` can be handled in a future PR.